### ISSUE TITLE
The file scanner/monitor is bad

### ISSUE THREAD

**Issue by rmcauley** [rmcauley/rainwave#122]
The current multi-threaded approach to scanning causes a lot of headaches and collisions on file operations, especially if we try to rename a file for UTF-8 friendliness.

UTF-8 and cp1252 filenames need to be supported, we have to stop renaming files, and we have to move to single-threaded operations to handle the order of operations correctly.

We also need to make sure that updated/moved files operate on a bit of a timeout or a check to make sure the files are complete.  We have a user now who is using torrent syncing for establishing his music directory with friends and a lot of times files are incomplete.  On top of that, as it stands, any file that's created but is 0B (e.g. FTP/SFTP uploads) are not handled very well.

In short: the scanner is really terrible as it is right now and needs a complete rethink.

**Comment by williamjacksn** [comment]
We should think about abandoning any sort of file watch daemon model, and only scan on demand.

I know a full library scan usually takes a few minutes, at least for Rainwave's library size. If there were a way to specify some subset of the library to scan, we wouldn't have to worry about how long a full library scan takes.

We could also take a look at [scandir](https://pypi.python.org/pypi/scandir) (specifically the `os.walk()` speedups) to see if that speeds up full library scan times.

**Comment by rmcauley** [comment]
I'm not keen on the idea of abandoning file-monitoring.  I want to keep in place for DJ support which is Coming Soon(tm) - I want to make sure anyone outside the Rainwave.cc managerial staff who might DJ can add extra music promptly. (if you have questions on these plans or how this works, feel free to ask me outside this ticket :) )

We used to have scan-on-demand, and music managers would very frequently forget to hit the scan button.  One more part of the procedure these days (which now includes Power Hours and social media posts) isn't as big a burden, but any extra burden on staff needs to be carefully weighed.  I see this as a problem I/we can solve, which means I/we should do it for the volunteer staff to help make their jobs easier.

**Comment by rmcauley** [comment]
http://rainwave.cc/forums/viewtopic.php?f=27&t=6070

This is tracked on the forums now.
